src/pluggy/_callers.py

```diff
@@ -70,6 +69,11 @@ def _multicall(
             for hook_impl in reversed(hook_impls):
                 try:
                     args = [caller_kwargs[argname] for argname in hook_impl.argnames]
+                    kwargs = {
```

Comment:
note that adding this bit undoes a specifically chosen optimization (that we might want to reevaluate later)

Reply:
Thank you for pointing this out. I understand what you mean. However, the current design has a major limitation that prevents the passing of keyword arguments.

Could you please elaborate on the specific optimization that is being undone by my addition? I would like to understand its purpose and how it was beneficial to the project.

Reply:
Thank you for sharing this information. Could you please elaborate on the performance evaluation method used? I would like to measure the performance based on the new version of Python to have a more accurate comparison. It would be helpful if you could provide more details on the testing environment .